File: Misc. Algos/GCD/stepdecrement.py
# Step decrementation algorithm
# Code by Akshat Aryan

def gcd(m, n):
    while(m != n):
        if(m > n):
            m = m - n
        else:
            n = n - m
    return m

if(__name__ == "__main__"):
    (a, b) = tuple(map(lambda x: int(x), input("Enter the two numbers:").split()))
    print("GCD =", gcd(a, b))
    

File: Misc. Algos/GCD/stepdecrement.c
// Step decrementation algorithm
// Code by Akshat Aryan
#include <stdio.h>

int gcd(int m, int n){
    while(m != n){
        if(m > n){
            m = m - n;
        }
        else{
            n = n - m;
        }
    }
    return m;
}

int main(){
    int a, b;
    printf("Enter the two numbers:");
    scanf("%d %d", &a, &b);
    printf("GCD = %d", gcd(a, b));
    return 0;
}
    